draft-ietf-tls-esni.md

```diff
@@ -917,6 +920,37 @@ and session IDs presented by the server. These connections are only used to
 trigger retries, as described in {{rejected-ech}}. This may be implemented, for
 instance, by reporting a failed connection with a dedicated error code.
 
+
+### Impact of Retry on Future Connections
+
+Clients MAY use information learned from a rejected ECH for future
```

Comment:
Wondering if this `MAY` should not be a stronger `SHOULD`. If not implemented by the client, it does mean that we have a doubling in the number of (expensive PKI involving) connections when ECH is disabled, in situations like TLS-terminating proxies, temporary disabling of ECH on a server, or indefinitely in case of a server config issue.

Reply:
If it's indefinitely, then the DNS records will time out. I think probably a SHOULD is too strong.